### PR TITLE
Enhancement LiveActivityFragment

### PR DESCRIPTION
**Pull request by elagin**
Show current / max heart rate. Ignore heart rate < 10... Gone button bar into LiveActivity. Auto scale totalStepsChart. Fix time x axis history chart. Draw total and per minute steps. Set limit on per minute chart by user settings.

2018-08-27-16-25-57-985gadgetbridge

### CODE REVIEW

**Comment by elagin** [comment]
How can I merge these three commits into one to get a clean story?

**Comment by Dikay900** [comment]
`git rebase -i HEAD~3`
then change 'pick' for the first two commits to 's' or 'squash'

**Comment by elagin** [comment]
So I do not see my commits
$ git rebase -i HEAD~4
pick ad440b09 Show current / max heart rate. Ignore heart rate < 10. Gone button bar into LiveActivity. Auto scale totalStepsChart. Fix time x axis history chart. Draw total and per minute steps. Set limit on p$
pick 915ad998 Added Scandinavian characters
pick 9fda67b2 Mi Band 3: Night Mode
pick 4d1ce2b7 Mi Band 3: temporarily disable sunset/sunrise night mode setting
.....

**Comment by Dikay900** [comment]
oh i see the double merge of the master branch did this mess to this branch.
for the future use `git rebase master` to merge new changes without messing up your history.

The easiest way is getting the changes for these single commits with `git format-patch master -o patches` from the live_activity branch and then redo the changes with a clean branch since this branch is not clean anymore.

You can also do `git reflog` and restore the branch with `git reset --hard sha1` before you did the merge then you have to add your latest change again.

You will find your commit with `git rebase -i HEAD~4` at the bottom but i think you cannot get the history clean anymore.

**Comment by cpfeiffer** [comment]
FWIW, there's no need to create a perfect history. You can always rebase your commits on top of master instead of merging master and force-push.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/charts/LiveActivityFragment.java [hunk 1]
```diff
@@ -233,10 +241,9 @@ private void addEntries(int timestamp) {
         }
         mHistorySet.addEntry(new Entry(timestamp, stepsPerMinute));
         int hr = getCurrentHeartRate();
-        if (hr < 0) {
-            hr = 0;
+        if (hr > HeartRateUtils.getInstance().getMinHeartRate()) {
+            mHeartRateSet.addEntry(new Entry(timestamp, hr));
         }
-        mHeartRateSet.addEntry(new Entry(timestamp, hr));
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/charts/LiveActivityFragment.java]: Does it look better if you skip entries with bad hr value?

Comment (elagin) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/charts/LiveActivityFragment.java]: Of course, why display garbage on the chart?

**Comment by elagin** [comment]
If there are no more objections on the code, then I will make a new branch and copy it there to create a separate PR.

**Comment by cpfeiffer** [comment]
Replaced by #1265

**Pull request closed without merging**